### PR TITLE
Fix Plotting Crash

### PR DESCRIPTION
**Pull request by AmrElsersy**
Fix Plotting Crash due to closing Plotting ign-gui plugin

@chapulina 
@claireyywang

### CODE REVIEW

File: src/PlottingInterface.cc [hunk 1]
```diff
@@ -80,19 +80,19 @@ class TransportPrivate
 class PlottingIfacePrivate
 {
   /// \brief Responsible for transport messages and topics
-  public: Transport *transport;
+  public: Transport transport;
 
   /// \brief current plotting time
-  public: double time;
+  public: double *time {new double};
```
Comment (chapulina) [inline src/PlottingInterface.cc:86]: Is this variable needed? Can we just have `plottingTimeRef`?

File: src/plugins/plotting/TransportPlotting.cc [hunk 2]
```diff
@@ -21,7 +21,8 @@ using namespace ignition;
 using namespace gui;
 using namespace plugins;
 
-TransportPlotting::~TransportPlotting() {}
+TransportPlotting::~TransportPlotting() {
```
Comment (chapulina) [inline src/plugins/plotting/TransportPlotting.cc:24]:
```suggestion
TransportPlotting::~TransportPlotting()
{
```

**Review by chapulina — CHANGES_REQUESTED** [review]

File: src/PlottingInterface.cc (hunk 1 above)
Comment (AmrElsersy) [reply src/PlottingInterface.cc:86]: yeah great idea :D 
i modified it

**Review by chapulina — APPROVED** [review]
LGTM! I just pushed a little change in d65abeed042004e8d417d70541b5afb5a11bfd5d to use `make_shared` instead of `new`, which is slightly more efficient.

**Pull request merged**